qpid/cpp/src/qpid/broker/amqp/Outgoing.cpp

```diff
@@ -79,6 +79,9 @@ OutgoingFromQueue::OutgoingFromQueue(Broker& broker, const std::string& source,
 
 void OutgoingFromQueue::init()
 {
+    // observe queue for changes to track undeliverable messages
+    queue->getObservers().add(boost::shared_ptr<QueueObserver>(shared_from_this()));
+
```

Comment:
I wondered about that. From other examples of QueueObserver's in the codebase I didn't see anything that explicitly removed itself from the observer list, so assumed (dangerously/lazily 😄) it was cleaned up automatically somewhere.